txp/Pos/Txp/MemState/Class.hs

```diff
@@ -100,7 +93,7 @@ modifyTxpLocalData reason f =
         lname <- getLoggerName
         liftIO . usingLoggerName lname $ txpMetricsWait reason
         timeBeginWait <- currentTime
-        (res, logMetricsRelease) <- withLock $ do
+        (res, logMetricsRelease) <- do
             timeEndWait <- currentTime
             liftIO . usingLoggerName lname $
                 txpMetricsAcquire (timeEndWait - timeBeginWait)
```

Comment:
We should move the code that checks how long we wait for the lock to the place where the lock is now taken -- otherwise, we'll always falsely report that we don't have to wait for the lock at all.
I'd rather not lose that information, in particular since we will increase usage of the `BlkSemaphore` with this PR.

Reply:
Hm, would it make sense to add move these metrics into the definition of `withBlkSemaphore`?

Reply:
Yes, so we have agreed that we will extend `withBlkSemaphore` with all the nice logging and metrics that we currently have here for the tx lock processing. It's ok to have the intermediate state where this PR is merged to master, so long as we do follow-up with the additions to `withBlkSemaphore`.

Reply:
I agree.